python/cuml/cuml/manifold/umap.pyx

```diff
@@ -749,6 +758,23 @@ class UMAP(Base,
             raise ValueError("Cannot provide a KNN graph when in \
             semi-supervised mode with categorical target_metric for now.")
 
+        # for getting n_rows of the dataset
+        _, self.n_rows, _, _ = \
+            input_to_cuml_array(X, order='C', check_dtype=np.float32,
+                                convert_to_dtype=(np.float32
+                                                  if convert_dtype
+                                                  else None))
```

Comment:
Can we hold on to merging this for a while? I added this part since we need
1. get number of rows
2. decide build algo given auto using number of rows
3. decide mem type given configured build algo (which should return the number of rows)

But I think calling `input_to_cuml_array` just for the sake of getting rows will add to mem usage. Let me figure this out before we merge this PR.

Reply:
Looks like `len` doesn't work on sparse inputs. Please use ``X.shape[0]`` instead, which should also work with all the inputs stated above (but also will work with sparse inputs)